Release 1.8 (formerly 1.7.1)

benjamn opened the pull request:
Now that Meteor 1.7.0.1 has been officially released and recommended (and [blogged about](https://blog.meteor.com/meteor-1-7-and-the-evergreen-dream-a8c1270b0901)), we need a place to begin collecting improvements for the next release, Meteor 1.8.

Note: for most of its development, Meteor 1.8 was called Meteor 1.7.1, but we recently decided to make the version number better reflect the significance of the release, per this comment: https://github.com/meteor/meteor/pull/9942#issuecomment-420744917

As with previous Meteor beta testing cycles, whenever you see a commit added to this pull request with a message like `Bump package versions for 1.8-rc.n release.`, that means you can run `meteor update --release 1.8-rc.n` to update an existing app to the new version, or `meteor create --release 1.8-rc.n your-new-app` to create a fresh Meteor 1.8 app.

If you want to participate in the development of Meteor 1.8, you should check out the [`release-1.7.1` branch](https://github.com/meteor/meteor/tree/release-1.7.1), regenerate the dev bundle, and run the `meteor` script from a checkout:

```sh
cd path/to/meteor
git fetch origin
git checkout -t origin/release-1.7.1
git submodule update --init --recursive
scripts/generate-dev-bundle.sh
./meteor --help
```

See [these instructions](https://github.com/meteor/meteor/blob/devel/Development.md#running-from-a-git-checkout) for help with running Meteor from a checkout.

KoenLav commented:
I'm interested in helping with two major features:
- Two-way batching of DDP messages;
- Refactoring (small parts of) the various classes involved in translating oplog notifications to collection observe callbacks (and eventually DDP messages).

I think the PR for batching DDP messages is basically good to go, but the refactoring of the way the oplog driver is implemented can be significantly reduced in size and improved upon. 

Would be much appreciated if someone of MDG could provide some feedback on the outstanding PRs.

KoenLav commented:
Also, in terms of future plans for Meteor, I would be really interested in working on an implementation of Meteor + React Native which allows using React components (rather than their native counterparts).

And I think meteor-desktop should be the officially recommended way to build desktop applications with Meteor, now it has reached version 1.0 (@wojtkowiak).

morloy commented:
I’d be interested in adding source maps for production builds. There’s already [zodern/minify-js-sourcemaps](https://github.com/zodern/minify-js-sourcemaps), but it needs to be updated for 1.7 and merged into master.

linegel commented:
There still around many people waiting for a way to use convenient Meteor API in react native apps with something like https://github.com/Urigo/meteor-client-bundler as @KoenLav mentioned, now react native plays key role in mobile development having even bigger market share then Cordova was at days of implementing inside meteor ecosystem. Adding possibility to use Meteor API inside react native apps even can create new big hype wave for Meteor and will make him truly cross-platform, as never before.

KoenLav commented:
@linegel there are already various (working) ways to link Meteor with React Native, but I am specifically interested in a way which allows Meteor to use React code for view rendering, rather than React Native code.

Because only then we can have a single source code which provides cross platform support and allows for interactions with the native side of things between Android, iOS, Windows, Mac and Linux.

linegel commented:
@KoenLav What are you asking is a quite hard task. With project linked above, you can have all logic the same across platforms, only visual components (or what you will also select) will be not ready to use across platforms. It's a task for react native team, to make framework ready to make even browser build in addition to currently supported platform, not for MDG. What is really missed, there almost two packages, https://github.com/inProgress-team/react-native-meteor which have not the same API and which have in production a lot of issues. The main issue of this project: it is not really re-used meteor code base. When the second one https://github.com/Urigo/meteor-client-bundler gives you the ability to even re-use same code for collections as was used for browser and so on. It's a big step into future, only one problem. It's not working for now :) 
Even example linked in readme not built properly :(

sakulstra commented:
I just tried updating to test #9983 , but suddenly our app stops building:

<details>
<summary>Log</summary>

```
=> Started proxy.
/home/lukass/.meteor/packages/coffeescript/.1.0.17.16ty2nn++os+web.browser+web.cordova/plugin.compileCoffeescript.os/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:116
throw error;
^

TypeError: Property right of AssignmentExpression expected node to be of a type ["Expression"] but instead got null
at Object.validate (/home/lukass/Documents/<project>/node_modules/@babel/core/node_modules/@babel/types/lib/definitions/utils.js:128:13)
at validate (/home/lukass/Documents/<project>/node_modules/@babel/core/node_modules/@babel/types/lib/validators/validate.js:17:9)
at builder (/home/lukass/Documents/<project>/node_modules/@babel/core/node_modules/@babel/types/lib/builders/builder.js:46:27)
at Object.AssignmentExpression (/home/lukass/Documents/<project>/node_modules/@babel/core/node_modules/@babel/types/lib/builders/generated/index.js:233:31)
at _core.types.sequenceExpression.identDecorators.map.decorator (/home/lukass/Documents/<project>/node_modules/@babel/plugin-proposal-decorators/lib/transformer-legacy.js:46:26)
at Array.map (<anonymous>)
  at applyEnsureOrdering (/home/lukass/Documents/<project>/node_modules/@babel/plugin-proposal-decorators/lib/transformer-legacy.js:43:59)
  at PluginPass.ClassExpression (/home/lukass/Documents/<project>/node_modules/@babel/plugin-proposal-decorators/lib/transformer-legacy.js:156:28)
  at newFn (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/visitors.js:193:21)
  at NodePath._call (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/path/context.js:53:20)
  at NodePath.call (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/path/context.js:40:17)
  at NodePath.visit (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/path/context.js:88:12)
  at TraversalContext.visitQueue (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:118:16)
  at TraversalContext.visitSingle (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:90:19)
  at TraversalContext.visit (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:146:19)
  at Function.traverse.node (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/index.js:94:17)
  at NodePath.visit (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/path/context.js:95:18)
  at TraversalContext.visitQueue (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:118:16)
  at TraversalContext.visitMultiple (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:85:17)
  at TraversalContext.visit (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:144:19)
  at Function.traverse.node (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/index.js:94:17)
  at NodePath.visit (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/path/context.js:95:18)
  at TraversalContext.visitQueue (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:118:16)
  at TraversalContext.visitQueue (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:124:21)
  at TraversalContext.visitMultiple (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:85:17)
  at TraversalContext.visit (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:144:19)
  at Function.traverse.node (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/index.js:94:17)
  at NodePath.visit (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/path/context.js:95:18)
  at TraversalContext.visitQueue (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:118:16)
  at TraversalContext.visitSingle (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:90:19)
  at TraversalContext.visit (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/context.js:146:19)
  at Function.traverse.node (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/index.js:94:17)
  at traverse (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/traverse/lib/index.js:76:12)
  at transformFile (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/core/lib/transformation/index.js:88:29)
  at runSync (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/core/lib/transformation/index.js:45:3)
  at Object.transformSync (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/@babel/core/lib/transform.js:43:38)
  at transform (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/meteor-babel/index.js:103:26)
  at compile (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/meteor-babel/index.js:113:5)
  at Cache.Cp.get (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/meteor-babel/cache.js:101:26)
  at Object.exports.compile (/home/lukass/.meteor/packages/ecmascript/.0.12.0-beta171.1.19c71oj.zxge++os+web.browser+web.browser.legacy+web.cordova/plugin.compile-ecmascript.os/npm/node_modules/meteor/babel-compiler/node_modules/meteor-babel/index.js:51:7)
  at Object.compile (packages/babel-compiler.js:44:29)
  at packages/babel-compiler.js:210:22
  at Function.time (/tools/tool-env/profile.js:305:10)
  at profile (packages/babel-compiler.js:258:20)
  at BabelCompiler.BCp.processOneFileForTarget (packages/babel-compiler.js:209:20)
  at packages/babel-compiler.js:123:25
  at _finalizerPromise.resolve (/tools/isobuild/compiler-plugin.js:880:28)
  at new Promise (<anonymous>)
    at JsOutputResource.finalize (/tools/isobuild/compiler-plugin.js:879:33)
    at JsOutputResource.reportPendingErrors (/tools/isobuild/compiler-plugin.js:889:10)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:822:14)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at each (/tools/isobuild/import-scanner.js:879:14)
    at _.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:87:22)
    at ImportScanner._scanFile (/tools/isobuild/import-scanner.js:843:5)
    at outputFiles.forEach.file (/tools/isobuild/import-scanner.js:503:14)
    at Array.forEach (<anonymous>)
      at ImportScanner.scanImports (/tools/isobuild/import-scanner.js:501:22)
      at sourceBatches.forEach.batch (/tools/isobuild/compiler-plugin.js:1232:17)
      at Array.forEach (<anonymous>)
        at Function.computeJsOutputFilesMap (/tools/isobuild/compiler-plugin.js:1200:19)
        at ClientTarget._emitResources (/tools/isobuild/bundler.js:1066:8)
        at buildmessage.enterJob (/tools/isobuild/bundler.js:837:12)
        at /tools/utils/buildmessage.js:359:18
        at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
        at /tools/utils/buildmessage.js:352:34
        at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
        at /tools/utils/buildmessage.js:350:23
        at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
        at Object.enterJob (/tools/utils/buildmessage.js:324:26)
        at ClientTarget.make (/tools/isobuild/bundler.js:828:18)
        at /tools/isobuild/bundler.js:3019:14
        at /tools/isobuild/bundler.js:3108:20
        at Array.forEach (<anonymous>)
          at Function._.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
          at /tools/isobuild/bundler.js:3107:7
          at /tools/utils/buildmessage.js:271:13
          at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
          at /tools/utils/buildmessage.js:264:29
          at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
          at /tools/utils/buildmessage.js:262:18
          at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
          at /tools/utils/buildmessage.js:253:23
          at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
          at Object.capture (/tools/utils/buildmessage.js:252:19)
          at bundle (/tools/isobuild/bundler.js:3000:31)
          at files.withCache (/tools/isobuild/bundler.js:2947:32)
          at Object.withCache (/tools/fs/files.js:1710:12)
          at Object.exports.bundle (/tools/isobuild/bundler.js:2947:16)
          at Profile.run (/tools/runners/run-app.js:579:36)
          at Function.run (/tools/tool-env/profile.js:490:12)
          at bundleApp (/tools/runners/run-app.js:578:34)
          at AppRunner._runOnce (/tools/runners/run-app.js:622:35)
          at AppRunner._fiber (/tools/runners/run-app.js:880:28)
          at /tools/runners/run-app.js:408:12
```

</details>

------------

I suspected it's related to the babel runtime update + decorators plugin, but removing that results in an other error:

<details>
<summary>Log</summary>

```
TypeError: Cannot read property 'substr' of undefined
    at Function.<anonymous> (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/source-map/lib/source-node.js:115:29)
    at Array.forEach (<anonymous>)
    at BasicSourceMapConsumer.SourceMapConsumer_eachMapping [as eachMapping] (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/source-map/lib/source-map-consumer.js:155:16)
    at Function.SourceNode_fromStringWithSourceMap [as fromStringWithSourceMap] (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/source-map/lib/source-node.js:80:26)
    at File.getPrelinkedOutput (/tools/isobuild/linker.js:751:38)
    at _.each.file (/tools/isobuild/linker.js:254:16)
    at Array.forEach (<anonymous>)
    at Function._.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
    at Module._buildModuleTree (/tools/isobuild/linker.js:232:7)
    at Module.getPrelinkedFiles (/tools/isobuild/linker.js:172:25)
    at Object.fullLink (/tools/isobuild/linker.js:1028:31)
    at buildmessage.enterJob (/tools/isobuild/compiler-plugin.js:1573:28)
    at /tools/utils/buildmessage.js:359:18
    at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:352:34
    at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:350:23
    at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
    at Object.enterJob (/tools/utils/buildmessage.js:324:26)
    at PackageSourceBatch._linkJS (/tools/isobuild/compiler-plugin.js:1572:18)
    at PackageSourceBatch.getResources (/tools/isobuild/compiler-plugin.js:1460:28)
    at sourceBatches.forEach.sourceBatch (/tools/isobuild/bundler.js:1090:37)
    at Array.forEach (<anonymous>)
    at ClientTarget._emitResources (/tools/isobuild/bundler.js:1072:19)
    at buildmessage.enterJob (/tools/isobuild/bundler.js:837:12)
    at /tools/utils/buildmessage.js:359:18
    at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:352:34
    at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:350:23
    at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
    at Object.enterJob (/tools/utils/buildmessage.js:324:26)
    at ClientTarget.make (/tools/isobuild/bundler.js:828:18)
    at /tools/isobuild/bundler.js:3019:14
    at /tools/isobuild/bundler.js:3108:20
    at Array.forEach (<anonymous>)
    at Function._.each._.forEach (/home/lukass/.meteor/packages/meteor-tool/.1.7.1-beta.1.5gzlbv.ow7l8++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
    at /tools/isobuild/bundler.js:3107:7
    at /tools/utils/buildmessage.js:271:13
    at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:264:29
    at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:262:18
    at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
    at /tools/utils/buildmessage.js:253:23
    at exports.EnvironmentVariable.withValue (/tools/utils/fiber-helpers.js:89:14)
    at Object.capture (/tools/utils/buildmessage.js:252:19)
    at bundle (/tools/isobuild/bundler.js:3000:31)
    at files.withCache (/tools/isobuild/bundler.js:2947:32)
    at Object.withCache (/tools/fs/files.js:1710:12)
    at Object.exports.bundle (/tools/isobuild/bundler.js:2947:16)
    at Profile.run (/tools/runners/run-app.js:579:36)
    at Function.run (/tools/tool-env/profile.js:490:12)
    at bundleApp (/tools/runners/run-app.js:578:34)
    at AppRunner._runOnce (/tools/runners/run-app.js:622:35)
    at AppRunner._fiber (/tools/runners/run-app.js:880:28)
    at /tools/runners/run-app.js:408:12
```

</details>


-----

Not sure if that helps anything, but i'll try to debug this a bit further when i'm more awake tomorrow :zzz:

benjamn commented:
@KoenLav While I too would love to see Meteor support React Native, perhaps we should wait and see what comes of [this rearchitecture](https://facebook.github.io/react-native/blog/2018/06/14/state-of-react-native-2018). The good news is that they say they're planning to make the bundler swappable!

copleykj commented:
I'm not sure full RN integration makes sense for Meteor. I guess maybe if you force the use of something like [react-native-web](https://github.com/necolas/react-native-web), but even then I think that its going to be more work than it's worth. [react-native-meteor](https://github.com/inProgress-team/react-native-meteor) allows RN apps to connect to your Meteor server pretty effortlessly and with minimal effort you can expose your apps API and share it between the two.

brucejo75 commented:
Is anyone looking at [meteor/validated-method](https://github.com/meteor/validated-method)?  There are some good [pull requests](https://github.com/meteor/validated-method/pulls) in there to keep it up to date.

jamesmillerburgess commented:
I just saw the 1.7 talk from the past Meteor Night and saw reference to the bundle visualizer issue. (https://github.com/meteor/meteor/pull/9559#issuecomment-360717928)

I meant to make a PR for this way back in January, but lost track of it. Sorry!

I'll make a simple one this week for the labels and later look at improving the UI, so hopefully it will no longer be an issue in 1.7.1.

jamesmillerburgess commented:
I just ran a project with `1.7.1-beta.8` and it is pulling in bundle-visualizer `1.2.1`, which still uses http. Do we need to bump bundle-visualizer to `1.7.1-beta.8` to get it to work with these beta releases?

benjamn commented:
@jamesmillerburgess Since `bundle-visualizer` is not a core package, we can just release it whenever: https://github.com/meteor/meteor/pull/10025#issuecomment-401397590

benjamn commented on packages/caching-compiler/caching-compiler.js:
```diff
@@ -281,12 +281,12 @@ CachingCompiler = class CachingCompiler extends CachingCompilerBase {
     const arches = this._cacheDebugEnabled && Object.create(null);
 
-    return Promise.all(inputFiles.map(async (inputFile) => {
+    inputFiles.forEach(inputFile => {
```
I'm honestly kind of angry how much of a performance difference switching from `Promise.all(async (inputFile) => ...)` to `inputFiles.forEach(inputFile => ...)` makes, here.

benjamn commented on tools/isobuild/bundler.js:
```diff
@@ -1804,6 +1772,65 @@ class ClientTarget extends Target {
+}, {
+  makeCacheKey(files, { arch, minifier, minifyMode }) {
+    return defaultMakeCacheKey(
+      minifier,
```
The caching system uses the `minifier` object as a key in a `WeakMap` behind the scenes, so the cache will be invalidated and its memory will be reclaimed whenever a new minifier plugin is constructed. Fortunately, that doesn't happen on every rebuild, unless the implementation of the minifier has changed, requiring it to be rebuilt. In other words, rarely.

sakulstra commented:
not sure if i messed sth up, or if there's a bug in the new bundle-visualizer. I just upgraded from beta 5 to 13 and the bundle shrinked a lot according to bundle-visuaizer, but it's also missing the entry for node_modules. E.g. `moment-timezone` which was a major slice in the chart is now gone.

benjamn commented:
Hmm, could be related to the minifier change: #10045

benjamn commented:
Here's the code where we extract bundle size information from the minified code: https://github.com/meteor/meteor/blob/devel/packages/standard-minifier-js/plugin/stats.js

kkukielka commented:
Is anyone else gets warning:
```
Warning: Nested Profile.run at sha512
Warning: Nested Profile.run at sha1
```
when `METEOR_PROFILE` env is exported?

(I'm using version 1.7.1-beta.15)

benjamn commented:
@kkukielka I've seen that, and I fixed it on another branch, but I just cherry-picked cacace3bcb548dcd69683916b1a1f14d888b627f to fix it here (when the next beta gets released).

benjamn commented:
171 commits so far in Release 1.7.1; methinks an RC must be imminent 🔮

StorytellerCZ commented:
@benjamn What about #9558 and other modernize efforts? Seems like there has been a stall there. Would be nice to finish what has already been started and reap the benefits.

lmachens commented:
I get this error on server when I change code (rebuild):

```
Errors prevented startup:

While running post-startup callbacks:
error: write EPIPE

Your application has errors. Waiting for file change.
C:\Users\Admin\AppData\Local\.meteor\packages\meteor-tool\1.7.1-beta.19\mt-os.windows.x86_64\dev_bundle\server-lib\node_modules\fibers\future.js:280
						throw(ex);
						^

TypeError: Cannot read property 'manifest' of undefined
    at RoutePolicy.checkForConflictWithStatic (packages/routepolicy/routepolicy.js:67:46)
    at RoutePolicy.declare (packages/routepolicy/routepolicy.js:95:12)
    at new StreamServer (packages/ddp-server/stream_server.js:38:15)
    at new Server (packages/ddp-server/livedata_server.js:1355:24)
    at server_convenience.js (packages/ddp-server/server_convenience.js:6:17)
    at fileEvaluate (packages\modules-runtime.js:322:7)
    at Module.require (packages\modules-runtime.js:224:14)
    at require (packages\modules-runtime.js:244:21)
    at C:\TrophyHunter\lol-trophy-hunter\.meteor\local\build\programs\server\packages\ddp-server.js:2227:1
    at C:\TrophyHunter\lol-trophy-hunter\.meteor\local\build\programs\server\packages\ddp-server.js:2234:3
    at C:\TrophyHunter\lol-trophy-hunter\.meteor\local\build\programs\server\boot.js:411:36
    at Array.forEach (<anonymous>)
    at C:\TrophyHunter\lol-trophy-hunter\.meteor\local\build\programs\server\boot.js:220:19
    at C:\TrophyHunter\lol-trophy-hunter\.meteor\local\build\programs\server\boot.js:471:5
    at Function.run (C:\TrophyHunter\lol-trophy-hunter\.meteor\local\build\programs\server\profile.js:510:12)
    at C:\TrophyHunter\lol-trophy-hunter\.meteor\local\build\programs\server\boot.js:470:11
```

Windows 10, Meteor 1.7.1-beta.19

lmachens commented:
@benjamn thank you, beta.20 fixed `TypeError: Cannot read property 'manifest' of undefined` issue.

lmachens commented:
@benjamn a new error on Windows with beta-20

```
I20180720-18:36:56.846(2)? Exception from task TypeError: Cannot destructure property `unpause` of 'undefined' or 'null'.
I20180720-18:36:56.847(2)?     at syncQueue.runTask (packages/webapp/webapp_server.js:599:27)
I20180720-18:36:56.847(2)?     at runWithEnvironment (packages\meteor.js:1356:24)
I20180720-18:36:56.848(2)?     at Object.task (packages\meteor.js:1369:14)
I20180720-18:36:56.848(2)?     at Meteor._SynchronousQueue.SQp._run (packages\meteor.js:987:16)
I20180720-18:36:56.849(2)?     at packages\meteor.js:964:12
=> Errors prevented startup:

   While running post-startup callbacks:
   error: Cannot destructure property `unpause` of 'undefined' or 'null'.

=> Your application has errors. Waiting for file change.
```

benjamn commented:
@lmachens Have you done something to disable the `web.browser.legacy` architecture? From both of these bug reports, it seems like `WebApp.clientPrograms["web.browser.legacy"]` is never created initially for your app.

lmachens commented:
Not that I know (actually I don't need it because it is a Chromium based project). 
package.json:

```
"meteor": {
    "mainModule": {
      "client": "client/index.js",
      "server": "server/index.js"
    }
  },
```

benjamn commented:
If there's any way to provide a small repro, I would love to try it out.

lmachens commented:
@benjamn I created a new project with `meteor create`. 
it happens when you change server code. if you change client code, it shows the delayed build message:

```
=> Started proxy.
=> Started MongoDB.
=> Started your app.

=> App running at: http://localhost:3000/
   Type Control-C twice to stop.

=> Client modified -- refreshing
=> Finished delayed build of web.browser.legacy in 398ms
I20180720-19:50:38.115(2)? test 123
=> Meteor server restarted
I20180720-19:50:38.227(2)? Exception from task TypeError: Cannot destructure property `unpause` of 'undefined' or 'null'.
I20180720-19:50:38.230(2)?     at syncQueue.runTask (packages/webapp/webapp_server.js:599:27)
I20180720-19:50:38.230(2)?     at runWithEnvironment (packages\meteor.js:1356:24)
I20180720-19:50:38.231(2)?     at Object.task (packages\meteor.js:1369:14)
I20180720-19:50:38.231(2)?     at Meteor._SynchronousQueue.SQp._run (packages\meteor.js:987:16)
I20180720-19:50:38.232(2)?     at packages\meteor.js:964:12
=> Errors prevented startup:

   While running post-startup callbacks:
   error: Cannot destructure property `unpause` of 'undefined' or 'null'.

=> Your application has errors. Waiting for file change.
```

benjamn commented:
@lmachens Ok, I have a local reproduction. Thanks for the heads up!

benjamn commented:
@lmachens Give `meteor update --release 1.7.1-beta.21` a try when you have a chance!

veered commented:
Sorry to bother you, but is there a potential ETA on this release? It'll help help with quite a few issues we are having in production (since we can't use 1.7.x until the rebuild improvements are released).

benjamn commented:
I guess it depends on whether we wait for Node 8.12.0… thoughts?

sakulstra commented:
As there's some cool stuff inside 8.12, but also some pretty big improvements in 1.7.1 couldn't we just create a patch release once 8.12 is ready without delaying 1.7.1?

veered commented:
Also, the version of Node used in production isn't tied to the Meteor release. The bundled code will just use whatever version of Node is installed on the machine. So I don't think there's too much of a tradeoff. I guess, the main issue would just be the asymmetry in versions between dev and prod.

jamesmillerburgess commented:
I just want to reiterate that I don't think the accounts changes have been fully QCed and there are likely more issues like #10094 lurking around in there.

Any testing we can get on the betas and release candidates from users of accounts will help a great deal towards getting this to a release-able state!

morloy commented:
When using the beta release and `WebAppInternals.enableSubresourceIntegrity();`, the resulting hash for `modules.js` is wrong. Other files seem to be ok. Any ideas?

benjamn commented:
@morloy Thanks for testing that. If the hash is wrong for `modules.js` in development, I would also be worried that it would be wrong for the combined JS bundle in production. How are you checking SRI correctness? Is there a particular browser/version that you're using?

morloy commented:
@benjamn I downloaded some JS files and computed the SRI manually on the command line. It was wrong for `modules.js` and correct for `meteor.js`. Haven’t tested with the production bundle.

jurgenfink commented:
Excellent @benjamn and Team
Release 1.7.1-beta.**31** no more error message as observed by @leosco and this humbled servant.

> ... To use the new parser, pass option { useNewUrlParser: true } to MongoClient.connect. ..

Many thanks and congratulations - keep up the good work :+1: 
You guys rock :1st_place_medal: :1st_place_medal:

arggh commented:
I was trying out the newest Meteor beta release and toying around with [meteor-svelte](https://github.com/meteor-svelte) by @klaussner which seemed to work nicely with Meteor `1.7.0.4`, but after upgrading to `1.7.1-beta.31` I got the error below in the client's console. The app was built without errors.

```
Uncaught TypeError: module.watch is not a function
    at App.html (app.js?hash=3f7c256e4f3ce1f987c65bacfe39918a1d4dce8d:14)
    at fileEvaluate (modules-runtime.js?hash=7f6bef46e4c39734b4c025bc1a8a760520295f53:332)
    at Module.require (modules-runtime.js?hash=7f6bef46e4c39734b4c025bc1a8a760520295f53:234)
    at Module.moduleLink [as link] (modules.js?hash=3f5faed87f3f76d969c5067eea35b898ac43f8df:272)
    at main.js (main.js:1)
    at fileEvaluate (modules-runtime.js?hash=7f6bef46e4c39734b4c025bc1a8a760520295f53:332)
    at Module.require (modules-runtime.js?hash=7f6bef46e4c39734b4c025bc1a8a760520295f53:234)
    at require (modules-runtime.js?hash=7f6bef46e4c39734b4c025bc1a8a760520295f53:254)
    at app.js?hash=3f7c256e4f3ce1f987c65bacfe39918a1d4dce8d:662
```

### Steps to reproduce

1. Clone https://github.com/meteor-svelte/ssr-example
2. Run `meteor  update --release 1.7.1-beta.31`
3. Run `npm install`
4. Run `npm install --save @babel/runtime@latest`
5. Run `meteor` and open your browser's console on `localhost:3000`

benjamn commented:
@arggh Can you copy/paste that into a new issue? I know what the problem is, but I'd like to discuss some options for solving it in a separate thread.

arggh commented:
Sure thing, referencing it here: https://github.com/meteor/meteor/issues/10148

hexsprite commented on .circleci/config.yml:
```diff
@@ -113,12 +113,25 @@ jobs:
+      - run:
+          name: Combine NPM Shrinkwrap Files
+          command: |
+            for d in packages/*/.npm/package; do cat $d/npm-shrinkwrap.json >> shrinkwraps.txt; done
```
couldn't this just be `cat packages/*/.npm/package packages/*/.npm/plugin/* >shrinkwraps.txt` ?

benjamn replied:
Yeah I think so!

benjamn commented:
Meteor 1.7.1 has now moved into the **release candidate** phase, so please try updating your apps with `meteor update --release 1.7.1-rc.0`, and report (or re-report) any problems that you encounter!

If you haven't been paying attention to this PR recently, this release is mostly about improving rebuild performance through lazy compilation and other optimizations. I hope those benefits are abundantly apparent, though [`History.md`](https://github.com/meteor/meteor/blob/release-1.7.1/History.md#v171-tbd) is also definitely worth a read, since we've added a ton of other features along the way.

Unless [Node 8.12.0](https://github.com/nodejs/node/pull/21593) is finalized very soon, and includes my [back-port](https://github.com/nodejs/node/pull/22122) of a fix for #9275, we will wait until Meteor 1.7.2 to update Node. Those changes by themselves are a big enough improvement that we should be able to release Meteor 1.7.2 quickly, whenever Node 8.12.0 happens, without waiting on anything else.

As usual, RC status means we won't add any new features before the final release of Meteor 1.7.1, but bugs will be fixed, with special attention to anything that eases updating to Meteor 1.7.1 from a previous Meteor version.

Thanks for your patience with this process, everyone!

arggh commented:
I was going to report `akryum:vue-component` being broken with the latest release, but apparently it's been broken since `1.7` as I just found these issues:

https://github.com/meteor-vue/vue-meteor/issues/326
https://github.com/meteor/meteor/issues/9946

I'll still point this out here before lots of Vue users upgrade to `v1.7.1`: maybe this is fastest track to get Vue up and running with Meteor again.

vooteles commented:
@arggh , for me Meteor 1.7.0.3 and vue-component run side by side just fine right now (have not tried 1.7.1 yet). From your links it seems likely that 1.7 introduced some compatibility issues between the two, but I'll just point out that it does not affect everyone.

hluz commented:
akryum:vue-component with 1.7.0.3 works fine for me, but upgrading to 1.7.1-rc.0 fails miserably ( reload request  received loop as reported in https://github.com/meteor-vue/vue-meteor/issues/326) 
:-(

benjamn commented:
@arggh @vooteles @hluz Try this suggestion? https://github.com/meteor-vue/vue-meteor/issues/326#issuecomment-413546822

Batistleman commented:
I can't seem to update to rc2 (or rc1). (I'm not offline, nor am I behind a proxy...)

```
app> meteor update --release 1.7.1-rc.2
Unable to update package catalog (are you offline?)

If you are using Meteor behind a proxy, set HTTP_PROXY and HTTPS_PROXY environment variables or see
this page for more details:  https://github.com/meteor/meteor/wiki/Using-Meteor-behind-a-proxy

Meteor 1.7.1-rc.2: unknown release.
```

Any ideas?

> Sidenote, for me the rc's are working very - very smooth! (build times ~9-20s instead of ~70-120s)

morloy commented:
With `enableSubresourceIntegrity()` in development, the SHA-512 still seems wrong for `modules.js`. Works in production, thus not a huge issue.

p3pp8 commented:
Hello, i've faced some problem while building for release:

`
Errors prevented bundling:                    
While minifying app stylesheet:               
app/imports/ui/components/items/helpItem/helpItem.css:76:3: Unknown word
`

there is also a nodeJS error:

`
Node#moveTo was deprecated. Use Container#append.
`
I'm using Meteor 1.7.1 with Babel 7.0.0 final version, just update a moment ago.

Any help really appreciated!

CaptainN commented:
I upgraded to rc.5 and @babel@runtime@7.0.0 (final release) and I get this message:

```
The version of @babel/runtime installed in your node_modules directory
(7.0.0-rc.2) contains a breaking change which was introduced by
https://github.com/babel/babel/pull/8266. Please either downgrade by
running the following command:

  meteor npm install --save-exact @babel/runtime@7.0.0-beta.55

or update to the latest beta version of Meteor 1.7.1, as explained in
this pull request: https://github.com/meteor/meteor/pull/9942.
```

My guess is whatever check is in there is not catching the final release (without any `-rc...` suffix).

Another side note - the console outputs this text on Windows (I don't know if that's from meteor or not):
`Type Control-C twice to stop.` But that's not correct - on Windows you press Control-C once, then it asks if you'd like to terminate the batch job (Y/N).

p3pp8 commented:
@CaptainN It is, you can use the Babel 7.0.0-rc.X or its final release ONLY using Meteor 1.7.1-rc5 or you have to stay on version 1.7.0.5 using Babel 7.0.0-beta.55. 
I'm using Meteor 1.7.1-rc.5 with the final version of Babel 7.0.0 without any problem. 
Waiting for the final 1.7.1 version of Meteor! Great job guys!!!!

CaptainN commented:
@p3pp8 It works fine- my report was about the notice only - it should probably not put up that notice if you have 1.7.1-rc.5 and the final release of babel 7.0.0, since you have everything set up correctly at that point, but I still get the notice.

p3pp8 commented:
@CaptainN Right, i haven't that notice with 1.7.1-rc.5 and Babel 7 final release...

CaptainN commented:
@p3pp8 - hmm, maybe it's related to having multiple meteor tool versions, or just an issue on Windows? I'm definitely on 1.7.1-rc.5, and have the latest version of babel in node_modules.

filipenevola commented:
What do you think about creating two new sections for each release note text: *breaking changes* and *migration steps*? 

I think it's better than currently that we need to read everything to find these important parts. 

Can I submit a PR doing this for 1.7.1 release?

It would be like:
## Breaking changes 
- Mongo queries relying on undefined.... 

## Migration steps
- Update @babel/runtime....

benjamn commented:
@filipenevola Go for it!

SimonSimCity commented:
@filipenevola Just because you mentioned queries relying on `undefined` ... you don't mean this change: https://github.com/meteor/meteor/commit/ce3885b6df10f9f69f425af8698642a388efa6f2 because it was already introduced in Meteor `1.6.1-beta.19` 😉 

Maybe it's also worth noting it for older versions ... at least also for `1.7.0`.

filipenevola commented:
@SimonSimCity done, updated from 1.6.1 https://github.com/meteor/meteor/pull/10209/files

It's very good to see how rare are breaking changes from Meteor 😄 and now that is clear on History.MD 🎉

sakulstra commented:
just fyi - in case you have bcrypt v3 installed and getting pre-gyp errors with rc.10 you might want to downgrade bcrypt to 1.x or wait for https://github.com/kelektiv/node.bcrypt.js/pull/649 to land and be released.

KoenLav commented:
Successfully upgraded to Meteor 1.7.1-rc.11.

Some issues with dependencies; I recommend running npm-check-updates (https://www.npmjs.com/package/npm-check-updates) to see which deps might need an update.

Do watch out upgrading front end libraries like Materialize/Material-UI/Bootstrap (many have shifted major versions lately).

Also running meteor update --all-packages is recommended.

nick-gudumac commented:
Just upgraded to rc.11 and deployed to production and it seems like CPU Spikes got much worse, almost all servers are at max CPU, if I revert to a previous RC version (before node upgrade) it's running normally. Anyone else experiencing this?

<img width="640" alt="app_planable_io_containers___galaxy" src="https://user-images.githubusercontent.com/9248127/45414732-b7c6d680-b684-11e8-8b16-2cdec49edd38.png">

p3pp8 commented:
I've noticed a huge slowdown with rc.11, rc.10 was good!

benjamn commented:
Going back to Node 8.11.4 and taking our time updating to Node 8.12.0 in Meteor 1.7.2 is definitely an option, but I'd like to understand what's causing the slowness. Can one of you open a new issue @p3pp8 @nick-gudumac?

p3pp8 commented:
@benjamn Done!

benjamn commented:
Seeing as how we're past the 400 commit mark for this PR, and the rebuild performance improvements are pretty significant, and there are some changes like #10200 that make me worry the update from 1.7.0.x won't be totally seamless for everyone, **would anyone object to releasing these changes as Meteor 1.8 instead of 1.7.1**? I realize we fell into the same trap when Meteor 1.6.2 became Meteor 1.7, but I promise we won't push the release back because of the new version number. I'm just as eager to get it done as anyone!

klaussner commented on History.md:
```diff
@@ -198,7 +198,7 @@ N/A
 
 * Mongo has been updated to version 4.0.2 and the `mongodb` npm package
-  used by `npm-mongo` has been updated to version 3.1.1.
+  used by `npm-mongo` has been updated to version 3.1.6.
```
I believe this will cause deprecation warnings because [`Collection#insert` is deprecated since 3.1.2](https://github.com/mongodb/node-mongodb-native/commit/a5d0f1d7e1fc3c39ac81fa31e04f2433f86ac183#diff-a3644fc6e558da2bde71112d24ec75baR664).

benjamn replied:
A lot of other `Collection` methods, too. Can we disable the deprecation warnings somehow?

klaussner replied:
The driver uses Node's `util.deprecate` function. As far as I'm aware, we can only disable *all* deprecation warnings (with `process.noDeprecation = true`) but not just those for `mongodb`.

benjamn replied:
Fortunately, the `process.noDeprecation` check only happens once, when the function is first wrapped by `util.deprecate`, so we can temporarily set `process.noDeprecation = true` while importing the `mongodb` package: 2a8dea89349935ff0e0452e5d1a1f5b09df954ad

mrauhu commented:
@benjamn, can we hope for merging the modular DDP `stream-server` #10118 and the `stream-server-uws` #10120 into the Meteor 1.8 release?

sakulstra commented:
When upgrading to latest rc things seem to work, but i get an:
```
=> Started proxy.                             
The version of @babel/runtime installed in your node_modules directory 
(7.0.0) contains a breaking change which was introduced by 
https://github.com/babel/babel/pull/8266. Please either downgrade by 
running the following command:

  meteor npm install --save-exact @babel/runtime@7.0.0-beta.55

or update to the latest beta version of Meteor 1.7.1, as explained in 
this pull request: https://github.com/meteor/meteor/pull/9942.

The version of @babel/runtime installed in your node_modules directory 
(7.0.0) contains a breaking change which was introduced by 
https://github.com/babel/babel/pull/8266. Please either downgrade by 
running the following command:

  meteor npm install --save-exact @babel/runtime@7.0.0-beta.55

or update to the latest beta version of Meteor 1.7.1, as explained in 
this pull request: https://github.com/meteor/meteor/pull/9942.
```
warning inside the console while the release is pinned to `METEOR@1.8-rc.16`

jamesmillerburgess commented:
@etyp Thanks for finding that - can you put it in a new issue?

I did much of the accounts modernization, so I'll try to take a look at this on Sunday when I will have access to a Windows machine.

etyp commented:
@jamesmillerburgess sure thing! I opened #10231 to track it. If you need to test IE, we usually use the [free IE VMs provided by microsoft](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/).

Thanks :)

veered commented:
@benjamn Any idea when 1.8 might be released? No rush, but I'm curious.

benjamn commented:
@veered Here's the plan: `1.8-rc.17` will be the last release candidate, unless there are any major bugs reported. We will publish Meteor 1.8 with Node 8.11.4 this week, but also simultaneously put out a Meteor 1.8.1 release that uses Node 8.12.0. For the time being, only 1.8 will be "recommended," so that's what you'll get if you type `meteor update`. If we can track down the garbage collection performance issues, and they can be solved with some package updates, then we might also recommend Meteor 1.8.1 at a later date. Otherwise, we'll shoot for Meteor 1.8.1.1 or 1.8.2 as the next recommended release after Meteor 1.8.

veered commented:
Awesome, that's great news :)

smeijer commented:
@benjamn, not sure if I should mention here, or create an issue. As there hasn't been an official advertising (blog post) yet. But on debian (CircleCI) I'm receiving an 403 install error.

```bash
curl "https://install.meteor.com?release=1.8.0" | /bin/sh
Downloading Meteor distribution

curl: (22) The requested URL returned error: 403
Retrying download in 5 seconds...

curl: (22) The requested URL returned error: 403
Retrying download in 5 seconds...
```

StorytellerCZ commented:
@smeijer try using release=1.8

smeijer commented:
Today I learned... Thanks @StorytellerCZ!

I do think it should be fixed one day. So it's clear that `1.8.0` is being enforced, and it will not be `1.8.1` some day (soon). But it's working.

@benjamn, awesome work on the release!
